desc/builder/builder_test.go

```diff
@@ -455,6 +456,109 @@ func TestBuildersFromDescriptors_PreserveComments(t *testing.T) {
 	testutil.Eq(t, count, descCount)
 }
 
+func TestBuilder_PreserveAllCommentsAfterBuild(t *testing.T) {
+	files := map[string]string{"test.proto": `
+syntax = "proto3";
```

Comment:
I think this is a much bigger example than necessary. Maybe just one enum and value and one message and field?